kg_query_walkthrough.ipynb

```diff
@@ -0,0 +1,183 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Walk Through for Knowledge Graph Testing\n",
+    "This notebook outlines the basic process of submitting queries to MULTIVAC's semantic knowledge graph. \n",
+    "First, we set up the required imports and arguments for the test. This process can be performed all at once from the commandline as well:<br><br>\n",
+    "`python3 map_queries.py -d data -o models -g models/glove.42B.300d -r model -m transe`<br><br>\n",
+    "(threshold and num_top_rel are by default 0.1 and 10 respectively, but can also be set at the commandline with flags `-t` and `-n` "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from multivac.src.rdf_graph.map_queries import *\n",
+    "tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "args_dict = {'dir': 'data',\n",
+    "             'out': 'models',\n",
+    "             'glove': 'models/glove.42B.300d',\n",
+    "             'run': 'model',\n",
+    "             'model': 'transe',\n",
+    "             'threshold': 0.1,\n",
+    "             'num_top_rel': 10}\n",
+    "\n",
+    "args_dict['search'] = '/Users/ben_ryan/Documents/DARPA_ASKE/Phase_II/openke_experiments/query_tester.csv'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we load up the knowledge graph embedding model previously calculated. This embedding model allows us to assign probabilities to missing nodes or relationships in the knowledge graph proposed via submitted queries. Here we are using TransE, an approach which models relationships by interpreting them as translations operating on the\n",
+    "low-dimensional embeddings of entities. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "con = config.Config()\n",
+    "con.set_in_path(args_dict['dir']+os.path.sep)\n",
+    "con.set_work_threads(8)\n",
+    "con.set_dimension(100)\n",
+    "con.set_test_link_prediction(True)\n",
+    "con.set_test_triple_classification(True)\n",
+    "\n",
+    "files = glob.glob(os.path.join(args_dict['out'],'*tf*'))\n",
+    "times = list(set([file.split('.')[2] for file in files]))\n",
+    "ifile = max([datetime.strptime(x, '%d%b%Y-%H:%M:%S') for x in times]).strftime('%d%b%Y-%H:%M:%S')\n",
+    "con.set_import_files(os.path.join(args_dict['out'], 'model.vec.{}.tf'.format(ifile)))\n",
+    "\n",
+    "con.init()\n",
+    "kem = set_model_choice(args_dict['model'])\n",
+    "con.set_model(kem)\n",
+    "\n",
+    "\n",
+    "files = [x for x in os.listdir(con.in_path) if '2id' in x]\n",
+    "rel_file = sorted([(os.path.getmtime(os.path.join(con.in_path, x)), x)\n",
```

Comment:
I made a comment in another file that this could be made a function; it'll help clean up the work here, especially for a notebook.